### PR TITLE
GET endpoint in Daksha : Allows users to retrieve TestResultData by providing TestUUID

### PR DESCRIPTION
**Pull request by Prashastmykaarma**
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have added tests to cover my changes.
- [x] My change requires a change to the documentation.
- [ ] I have Extended README or added documentation/comments where applicable
- [x] Added copyright headers for new files from CONTRIBUTING.md
- [x] My code follows the code style of this project.

- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have updated the [ChangeLog](CHANGELOG.md) following [Semantic Versioning](https://semver.org)

### CODE REVIEW

File: engine/views.py [hunk 1]
```diff
@@ -62,6 +63,62 @@ def executor(request):
...
+    """
+    errors=[]
+    testresults=[]
+    if request.method == 'GET':
+        try:
+            logger.info("GET request recieved")
+            if(TEST_RESULT_DB!= None and TEST_RESULT_DB.lower() == 'postgres'):
+                results_for_given_testuuid=TestResults.objects.all().filter(TestUUID=testuuid).values()
+                if request.body and results_for_given_testuuid:
+                    testnames=json.loads(request.body)
```
Comment (hriddhighosh4) [inline engine/views.py:79]: log

File: engine/views.py [hunk 2]
```diff
@@ -62,6 +63,62 @@ def executor(request):
...
+    Receives GET request and returns relevant data from the database
+
+    """
+    errors=[]
+    testresults=[]
+    if request.method == 'GET':
+        try:
+            logger.info("GET request recieved")
+            if(TEST_RESULT_DB!= None and TEST_RESULT_DB.lower() == 'postgres'):
+                results_for_given_testuuid=TestResults.objects.all().filter(TestUUID=testuuid).values()
```
Comment (hriddhighosh4) [inline engine/views.py:77]: test_results_for_uuid

File: engine/views.py (hunk 1 above)
Comment (Prashastmykaarma) [reply engine/views.py:79]: Done

File: engine/views.py (hunk 2 above)
Comment (Prashastmykaarma) [reply engine/views.py:77]: Done

File: engine/views.py [hunk 3]
```diff
@@ -62,6 +63,95 @@ def executor(request):
...
+def retriever(request, testuuid):
+    """
+    Receives GET request and returns relevant data from the database
+
+    """
+    errors = []
+    testresults = []
+    if request.method == "GET":
+        try:
+            logger.info("GET request recieved")
```
Comment (hriddhighosh4) [inline engine/views.py:75]: Fetching test results from db

File: engine/models.py [hunk 4]
```diff
@@ -37,6 +37,10 @@ def __init__(self, name, test_status, step, failure_reason):
         self.failed_step = step
         self.failure_cause = failure_reason
 
+class GetTestResults:
```
Comment (hriddhighosh4) [inline engine/models.py:40]: GetTestResultsResponse

File: engine/views.py [hunk 5]
```diff
@@ -62,6 +63,95 @@ def executor(request):
...
+                    )
+                    fetched_test_results_json = json.loads(
+                        fetched_test_results_json_string
+                    )
+                    return JsonResponse(
+                        fetched_test_results_json, status=status.HTTP_400_BAD_REQUEST
+                    )
+
+                if request.body:
+                    testnames = json.loads(request.body)
```
Comment (hriddhighosh4) [inline engine/views.py:96]: User has opted for specific test results in the test uuid

File: engine/models.py (hunk 4 above)
Comment (Prashastmykaarma) [reply engine/models.py:40]: Done

File: engine/views.py (hunk 5 above)
Comment (Prashastmykaarma) [reply engine/views.py:96]: Done

File: engine/views.py [hunk 6]
```diff
@@ -62,6 +63,95 @@ def executor(request):
...
+    testresults = []
+    if request.method == "GET":
+        try:
+            logger.info("GET request recieved")
+            if TEST_RESULT_DB != None and TEST_RESULT_DB.lower() == "postgres":
+                test_results_for_uuid = (
+                    TestResults.objects.all().filter(TestUUID=testuuid).values()
+                )
+                if not test_results_for_uuid:
+                    errors.append(
```
[the remark this answers is not in the document]
Comment (Prashastmykaarma) [reply engine/views.py:81]: Done

File: engine/views.py (hunk 3 above)
Comment (Prashastmykaarma) [reply engine/views.py:75]: Done

**Review by hriddhighosh4 — APPROVED** [review]

**Pull request merged**